R/hugo.R

```diff
@@ -21,17 +21,23 @@ hugo_version = function() {
 #' @param local Whether to build the site for local preview (if \code{TRUE}, all
 #'   drafts and future posts will also be built, and the site configuration
 #'   \code{baseurl} will be set to \code{/} temporarily).
+#' @param minify Whether to use minification on the final rendered output (if \code{TRUE},
+#'   \code{--minify} is added to the call to \code{hugo}). Supported for hugo versions >= 0.47.
+#'   Use \code{options(blogdown.minify = TRUE)} to enable minification in non-direct
+#'   calling context.
 #' @export
 #' @describeIn hugo_cmd Build a plain Hugo website. Note that the function
 #'   \code{\link{build_site}()} first compiles Rmd files, and then calls Hugo
 #'   via \code{hugo_build()} to build the site.
-hugo_build = function(local = FALSE) {
+hugo_build = function(local = FALSE, minify = getOption("blogdown.minify")) {
   config = load_config()
   # Hugo 0.48 generates an ugly empty resources/ dir in the root dir
   on.exit(bookdown:::clean_empty_dir('resources'), add = TRUE)
   if (local) tweak_hugo_env()
+  do_minify = isTRUE(minify) && isTRUE(can_minify())
   hugo_cmd(c(
     if (local) c('-b', site_base_dir(), '-D', '-F'),
+    if (isTRUE(do_minify)) c('--minify'),
```

Comment:
Let's just do this:

```suggestion
    getOption('blogdown.hugo.args'),
```

since it is more general-purpose and allows you to set arbitrary hugo options, not limited to `--minify`.

This new option should be documented at https://bookdown.org/yihui/blogdown/more-global-options.html

Reply:
Thanks, done in https://github.com/jozefhajnala/blogdown/commit/db3c0da794de7eefcefb3847fa9b596b67d7e83d